WooCommerce/Classes/Authentication/AppleIDCredentialChecker.swift

```diff
@@ -0,0 +1,104 @@
+import KeychainAccess
+import WordPressAuthenticator
+
+/// Checks and listens for observations when the Apple ID credential is revoked when the user previously signed in with Apple.
+///
+@available(iOS 13.0, *)
+final class AppleIDCredentialChecker {
+    /// Keychain access for SIWA auth token
+    private lazy var keychain = Keychain(service: WooConstants.keychainServiceName)
+
+    private var cancellable: ObservationToken?
+
+    private lazy var authenticator: WordPressAuthenticator = WordPressAuthenticator.shared
```

Comment:
actually, after observing `didBecomeActive` inside the checker class https://github.com/woocommerce/woocommerce-ios/pull/2818/commits/01bce49859e57d5592a8027e82bac0d46be49b87, this lazy var change can be reverted (https://github.com/woocommerce/woocommerce-ios/pull/2818/commits/e3c73b13968fc096cbc2eb9bed91264bad8c3efc) since it's always after the class is initialized and after `WordPressAuthenticator` singleton init 😄